src/Symfony/Component/Messenger/Asynchronous/Routing/SenderLocatorInterface.php

```diff
@@ -25,7 +26,16 @@
      *
      * @param object $message
      *
-     * @return SenderInterface[]
+     * @return SenderInterface|null
      */
-    public function getSendersForMessage($message): array;
+    public function getSenderForMessage($message): ?SenderInterface;
+
+    /**
+     * Whether to route the message to a sender and still have it passed to its respective handler.
+     *
+     * @param object $message
+     *
+     * @return bool
+     */
+    public function forwardToSenderAndHandler($message): bool;
```

Comment:
This doesn't seem to be the `SenderLocator`'s responsibility. We should simply move the mapping behind this method to the `SendMessageMiddleware` IMHO.

Reply:
Good point. Moved it to the middleware.